website/docs/developer/research/research-orca-delete-functionality.mdx

```diff
@@ -0,0 +1,179 @@
+    The api gateway resource policy can be used to restrict certain users from accessing the api endpoint. For more information, see [Controlling access to an API with API Gateway resource policies](https://docs.aws.amazon.com/apigateway/latest/developerguide/apigateway-resource-policies.html).
+    Another technology to look into would be AWS Cognito which can be used to create customizable authentication and authorization solutions for the REST API. Amazon Cognito user pools are used to control who can invoke REST API methods. For more information, see [Control access to a REST API using Amazon Cognito user pools as authorizer](https://docs.aws.amazon.com/apigateway/latest/developerguide/apigateway-integrate-with-cognito.html).
+
+:::note
+Authentication and authorization piece of this functionality needs further research to make sure only authorized users can perform deletes.
+:::
+
+:::warning
+Currently AWS Cognito has some restrictions from use in production environments and should be avoided for now.
+:::
+
+
+- As a DAAC, I would like duplicate data to be versioned in ORCA in order to recover the proper version of the data based on the type of failure.	
+
+    Versioning can be enabled at the bucket level by configuring the bucket properties via terraform or AWS console. After versioning is enabled in a bucket, it can never return to an unversioned state but can only be suspended on that bucket. Versioning can help recover objects from accidental deletion or overwrite. For example, if an object is deleted, Amazon S3 inserts a delete marker instead of removing the object permanently. The delete marker becomes the current object version. If the object is overwritten, it results in a new object version in the bucket. The previous version can always be restored. For more information, see [Deleting object versions from a versioning-enabled bucket](https://docs.aws.amazon.com/AmazonS3/latest/userguide/DeletingObjectVersions.html).
+    The drawback of using versioning is the additional cost of storage for that versioned object. This can be solved by using object versioning with S3 Lifecycle policy. After a certain period of time defined, the versioned object will be marked as expired and automatically deleted from S3 bucket. This can be configured in bucket management via terraform. For more information, see [S3 lifecycle policy](https://docs.aws.amazon.com/AmazonS3/latest/userguide/object-lifecycle-mgmt.html)
+    
+
+
+- Perform a check with CMR and/or Cumulus to verify that the granule has been deleted from those locations prior to allowing the delete to proceed in ORCA.
+
+    
+    This can be implemented in the existing lambda function that will be used to delete the data from ORCA.
+    Before deleting from CMR and Cumulus, The UI should show a warning message to the user for additional confirmation.
```

Comment:
I would add a line stating that additional items need to be discussed on this rule to determine when and how it would be implemented and to address corner cases like versioned file removal.

Reply:
added